### DISCUSSION TITLE
Any way to 'calculate' bw savings?

### DISCUSSION THREAD

**Discussion by inside83** [Novage/p2p-media-loader#201] [Q&A]
Hi,
I'm using p2p-media-loader with wt-tracker and it works great.
It would be even more awesome if there's some sort of statistics like 'total amount of data transferred via p2p compared to total amount of data transferred via HTTP'.
Only stats I get are the live one, by accessing `https://my.tracker.domain/stats.json`.

**Comment by mrlika** [comment]
This projects implements P2P and allows getting stats of each individual peer. Gathering, storing and creating reports from all the peers is called analytics. There are a lot of analytics services and open source solutions in the Web.

**Reply by inside83** [reply]
Thank you for the reply
Are there any analytics services you would recommend?
For example there are graphs on
https://novage.com.ua/p2p-media-loader/demo.html
so that must be from one of the analytics services you're mentioning, right?
I'm just interested in the most basic analytical data like number of peers (per torrent and in total) and amount of data served to and from peers in a given period.

Like I said, p2p-media-loader with wet-tracker is working great but I'd like to measure the it's benefits and know how much bandwidth is it saving.

**Reply by mrlika** [reply]
No. This demo shows statistics of a single peer that you can calculate using events.

Getting statistics of huge peer network requires analytics service. All the peers send stats to a server/service that saves it and then can combine everything in one report.

What I know is that Google Analytics is not very good for this purpose due to limitation in number of analytics data messages per user per session that doesn't allow you to let say send statistics from peer each minute during 2 hours video stream.

Check other available analytics services and their limitations or you can develop your own simple analytics endpoint.
